cpp-coverals will now raise an exception if neither the travis job id nor the repo token are found

mkfifo opened the pull request:
This addresses the initial work raised in issue #82 but in a way that works with the existing travis workflow (travis job id)

eddyxu commented on cpp_coveralls/__init__.py:
```diff
@@ -84,7 +84,13 @@ def run():
+
     cov_report = coverage.collect(args)
+
+    if not cov_report.get("repo_token") and not cov_report.get("service_job_id"):
```
Can you raise the `Error` after parsing argument? We don't need to run `gcov` before finding out one of these fields is missing.

mkfifo commented:
@eddyxu good point, thanks for that.

I have made a new push that should raise the error in the correct location (after arg parsing and before any gcov).

mkfifo commented:
re-pushed, had a silly mistake in my commit.

eddyxu commented on cpp_coveralls/__init__.py:
```diff
@@ -82,9 +82,15 @@ def run():
+
     if not args.no_gcov:
         coverage.run_gcov(args)
+
```
The commit looks very good. Just a few minor comments:

* Could you remove the blank line changes here.
* Could you rebase to merge you changes into one commit?

eddyxu commented on test.bash:
```diff
@@ -24,6 +24,7 @@ int main() {
 EOF
 gcc -coverage -o foo foo.c
 ./foo
+export COVERALLS_REPO_TOKEN="fake testing token"
```
Could you test whether the 'TRAVIS_JOB_ID` is set before set repo token?

We also want to test this logic on a travis machine.

mkfifo commented:
@eddyxu sorry about that, I was meaning to squash them down once I had all the kinks worked out.

I have pushed a new squashed fix, I have also updated the test script to work in both travis and non-travis environments, further I moved the COVERALLS_REPO_TOKEN handling code to be in __init__.py:run along with the yaml handling code (as it was in coverage.py:collect which is called too late).

I removed the whitespace as this commit is meant to be functional only, however stylistically I do dislike how tightly packed all the branching code in __init__.py:run is, especially considering the lack of comments as it makes for a very hard read.

eddyxu commented:
Cool, @mkfifo  Thanks a lot for working on this!
